### ISSUE TITLE
spi master gives output at all the four signals

### ISSUE THREAD

**Issue by ravi02011993** [pulp-platform/pulpino#198]
![image](https://user-images.githubusercontent.com/28788219/39420728-1e8ab8fc-4c66-11e8-9870-2e15a041e5ef.png)

Dear all,

Currently  I am working on SPI master to transfer the data into one particular address.
I have used the following commands for giving the output at sdo0.

spi_setup_dummy( 0 , 0);

  spi_setup_master(1); 
  spi_setup_cmd_addr(0x1b , 8 ,0x001050FF ,32);
  spi_set_datalen(120);
  spi_start_transaction(SPI_CMD_WR , SPI_CSN0);
spi_write_fifo( data , Datalen)

I have the following query

1. CS0 is only getting low and the other chip selects are not active. But the serial output is available
in all the four ports from sdo0- sdo3. Can someone tell me what is the reason for this?

Thanks in advance

Regards
Ravi

**Comment by ravi02011993** [comment]
![image](https://user-images.githubusercontent.com/28788219/39518755-878e0e4a-4e04-11e8-9f5e-552515e563f1.png)

@FrancescoConti 
i have configured the apb_spi_master peripheral in spi_std mode. please find the attached screenshot.
spi_mode is 0 but the output is available in all the 4pins.

**Comment by WaleedOsama** [comment]
Are you sure this is quad mode ? 

In quad mode data will be distributed on the four lines 

So if you send 4'b1101 in quad mode 

So sdo0 = 1 , sdo1 = 1 , sdo2 = 0 , sdo3 = 1

But in standard mode the 4 bits will be sent after each other on sd0.

But i noticed that in hardware of spi in standard mode they make the same output in sd0 on the other 3 lines , i do not know what is the use of such move .

but if you want standard mode , and you are seeing ( command , address , dummy , data ) on sdo0 then just disregard the other 3 output lines .

**Comment by ravi02011993** [comment]
Yeah, I also thought to disregard the remaining outputs. Thank you
